specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -5189,6 +5203,48 @@
       },
       "description": "Response from a List SynonymMaps request. If successful, it includes the full definitions of all synonym maps."
     },
+    "EncryptionKey": {
+      "properties": {
+        "keyVaultKeyName": {
+          "type": "string",
+          "description": "The name of the key, from your Azure KeyVault, you wish to use to protect your data. "
+        },
+        "keyVaultKeyVersion": {
+          "type": "string",
+          "description": "The specific version of the key, from your Azure KeyVault, you wish to use to protect your data."
+        },
+        "keyVaultUri": {
+          "type": "string",
+          "description": "The URI of the Azure KeyVault where you created the key you want to use to protect your data. This value is also sometime referred to as the KeyVault DNS Name. An example value would be : https://your-keyvault-name.vault.azure.net."
+        },
+        "accessCredentials": {
+          "$ref": "#/definitions/AzureActiveDirectoryApplicationCredentials",
+          "description": "The credentials to use to authenticate to the provided KeyVault. Can be left unspecified if MSI (Managed Service Identity) is enabled for the service.",
+          "externalDocs": {
+            "url": "https://aka.ms/azure-search-msi"
+          }
+        }
+      },
+      "required": [
+        "keyVaultKeyName",
+        "keyVaultKeyVersion",
+        "keyVaultUri"
+      ],
+      "description": "The configuration to use encryption with customer-managed keys. This information will be used to encrypt or decrypt the data you have protected using your own keys."
+    },
+    "AzureActiveDirectoryApplicationCredentials": {
+      "properties": {
+        "applicationId": {
+          "type": "string",
+          "description": "The is the ID of the Azure Active Directory Application that will be used to authenticate to your KeyVault. The Application ID is not to be confused with the Object ID of your Azure Active Directory Application."
+        },
+        "applicationSecret": {
+          "type": "string",
+          "description": "The application secret, also sometime refered to as the authentication key value. This will be used to authenticate to your KeyVault."
+        }
+      },
+      "description" : "The credentials of the Azure Active Directory Application identity we are going to use to authenticate to your KeyVault when we need to access your encryption keys."
```

Comment:
Avoid anthropomorphizing the service with "we":

"The credentials of the Azure Active Directory Application identity that Azure Search uses to authenticate to your KeyVault. This allows Azure Search to access your encryption key, which it uses to encrypt and decrypt your data."

Reply:
I like your version Bruce, but I think we need to be really when we're talking about a "managed identity" vs. "an application identity created for a registered application in AAD".   They are both identities in AAD.   The registered app process is confusing and complicated -- I can see why they are pushing "managed identities"

Reply:
I kept what @brjohnstmsft wrote, but added "registered" in front of "Azure Active Directory Application Identity" to disambiguiate

Reply:
"The credentials of the registered Azure Active Directory Application identity that Azure Search uses to authenticate to your Key Vault. This allows Azure Search to access your encryption key, which it uses to encrypt and decrypt your data."